File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    answer.append(arr[0])
    # O(n) 시간복잡도 가지는 게 좋을 듯
    for i in range(1,len(arr)):
        if answer[-1] == arr[i]:
            continue
        else:
            answer.append(arr[i])
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;
import java.io.*;

public class Solution {
    public Stack<Integer> solution(int []arr)  throws IOException {
        BufferedWriter bufferedWriter = new BufferedWriter(new OutputStreamWriter(System.out));
        Stack<Integer> stack = new Stack<>();

        for (int a: arr) {
            if (!stack.empty()) {
                if (!stack.peek().equals(a)) {
                    stack.push(a);
                }
            } else {
                stack.push(a);
            }
        }
        // while (!stack.empty()) {
        //     bufferedWriter.write(stack.pop());
        //     bufferedWriter.write("\n");
        //     bufferedWriter.flush();
        // }
        bufferedWriter.close();
        return stack;
    }
}